[SDL-0257] New vehicle data HandsOffSteering

yoooriii opened the pull request:
Fixes #1473 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR
- [x] I have run the unit tests with this PR

#### Unit Tests
Added tests to cover this functionality, tested on my local machine, iPhone-8 simulator running iOS-13.0.

#### Core Tests

### Summary
[detailed info on sdl evolution](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0257-New-vehicle-data-HandsOffSteering.md)

### Changelog
##### Breaking Changes

##### Enhancements

##### Bug Fixes

### Tasks Remaining:

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

yoooriii commented:
@atiwari9 could you please review?

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:
```diff
@@ -117,7 +119,9 @@
                                                                    SDLRPCParameterNameTurnSignal:vehicleDataResult,
-                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult},
+                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult,
+                                                                   @"handsOffSteering":vehicleDataResult
```
Use SDLRPCParameterNameHandsOffSteering

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:
```diff
@@ -153,10 +157,11 @@
     
     it(@"Should return nil if not set", ^ {
-        SDLSubscribeVehicleDataResponse* testResponse = [[SDLSubscribeVehicleDataResponse alloc] init];
+        SDLSubscribeVehicleDataResponse* testResponse = [SDLSubscribeVehicleDataResponse new];
```
`alloc init`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLUnsubscribeVehicleDataResponseSpec.m:
```diff
@@ -152,10 +155,11 @@
     
     it(@"Should return nil if not set", ^ {
-        SDLUnsubscribeVehicleDataResponse* testResponse = [[SDLUnsubscribeVehicleDataResponse alloc] init];
+        SDLUnsubscribeVehicleDataResponse* testResponse = [SDLUnsubscribeVehicleDataResponse new];
```
`alloc init`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:
```diff
@@ -147,8 +148,10 @@
-                                                     SDLRPCParameterNameWiperStatus:SDLWiperStatusAutomaticHigh},
-                                             SDLRPCParameterNameOperationName:SDLRPCFunctionNameGetVehicleData}} mutableCopy];
+                                                     SDLRPCParameterNameWiperStatus:SDLWiperStatusAutomaticHigh,
+                                                     @"handsOffSteering":@YES
```
Use `SDLRPCParameterNameHandsOffSteering`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -110,7 +112,9 @@
                                                            SDLRPCParameterNameTurnSignal:@YES,
-                                                           SDLRPCParameterNameWiperStatus:@YES},
+                                                           SDLRPCParameterNameWiperStatus:@YES,
+                                                           @"handsOffSteering":@YES
```
Use `SDLRPCParameterNameHandsOffSteering`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:
```diff
@@ -110,6 +111,7 @@
         expect(testNotification.turnSignal).to(equal(SDLTurnSignalRight));
         expect(testNotification.vin).to(equal(@"222222222722"));
         expect(testNotification.wiperStatus).to(equal(SDLWiperStatusStalled));
+        expect(testNotification.handsOffSteering).to(beTrue());
```
Use `expect(testNotification.handsOffSteering).to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:
```diff
@@ -184,6 +187,7 @@
         expect(testResponse.turnSignal).to(equal(SDLTurnSignalOff));
         expect(testResponse.vin).to(equal(vin));
         expect(testResponse.wiperStatus).to(equal(SDLWiperStatusAutomaticHigh));
+        expect(testResponse.handsOffSteering).to(beTrue());
```
Use `expect(testResponse.handsOffSteering).to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:
```diff
@@ -112,10 +112,11 @@
         expect(testResponse.turnSignal).to(equal(SDLTurnSignalBoth));
         expect(testResponse.vin).to(equal(vin));
         expect(testResponse.wiperStatus).to(equal(SDLWiperStatusAutomaticHigh));
+        expect(testResponse.handsOffSteering).to(beTrue());
```
Use expect(testResponse.handsOffSteering).to(equal(@YES));

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -16,7 +16,7 @@
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLGetVehicleData* testRequest = [[SDLGetVehicleData alloc] init];
+        SDLGetVehicleData* testRequest = [SDLGetVehicleData new];
```
`alloc init`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -16,7 +16,7 @@
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
`alloc init`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
 describe(@"initializers", ^{
     context(@"init", ^{
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
`alloc init`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:
```diff
@@ -181,6 +185,7 @@
         expect(testNotification.vin).to(equal(@"222222222722"));
         expect(testNotification.wiperStatus).to(equal(SDLWiperStatusStalled));
+        expect(testNotification.handsOffSteering).to(beTrue());
     });
```
Use `.to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -77,6 +78,7 @@
         expect(testRequest.tirePressure).to(equal(@NO));
         expect(testRequest.turnSignal).to(equal(@YES));
         expect(testRequest.wiperStatus).to(equal(@NO));
+        expect(testRequest.handsOffSteering).to(beTrue());
```
Use `.to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
         expect(testRequest.tirePressure).to(equal(@YES));
         expect(testRequest.turnSignal).to(equal(@NO));
         expect(testRequest.wiperStatus).to(equal(@YES));
+        expect(testRequest.handsOffSteering).to(beTrue());
```
Use `.to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
+            expect(testRequest.tirePressure).to(equal(@YES));
+            expect(testRequest.turnSignal).to(equal(@YES));
+            expect(testRequest.wiperStatus).to(equal(@YES));
+            expect(testRequest.handsOffSteering).to(beTrue());
```
Use `.to(equal(@YES));`

SatbirTanda commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
+            expect(testRequest.tirePressure).to(equal(@YES));
+            expect(testRequest.turnSignal).to(equal(@YES));
+            expect(testRequest.wiperStatus).to(equal(@YES));
+            expect(testRequest.handsOffSteering).to(beTrue());
```
Use `.to(equal(@YES));`

yoooriii commented:
Hello, @theresalech could you please review?

joeljfischer commented on SmartDeviceLink/SDLGetVehicleDataResponse.h:
```diff
@@ -1,190 +1,272 @@
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the SmartDeviceLink Consortium Inc. nor the names of
+ * its contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
 
-#import "SDLComponentVolumeStatus.h"
-#import "SDLElectronicParkBrakeStatus.h"
-#import "SDLPRNDL.h"
 #import "SDLTurnSignal.h"
-#import "SDLVehicleDataEventStatus.h"
+#import "SDLElectronicParkBrakeStatus.h"
 #import "SDLWiperStatus.h"
+#import "SDLVehicleDataEventStatus.h"
+#import "SDLPRNDL.h"
+#import "SDLRPCResponse.h"
+#import "SDLComponentVolumeStatus.h"
```
These were changed to be out of order. Please fix.

joeljfischer commented on SmartDeviceLink/SDLGetVehicleDataResponse.m:
```diff
@@ -1,24 +1,57 @@
-#import "SDLDeviceStatus.h"
-#import "SDLECallInfo.h"
 #import "SDLEmergencyEvent.h"
-#import "SDLFuelRange.h"
-#import "SDLGPSData.h"
-#import "SDLHeadLampStatus.h"
+#import "SDLTurnSignal.h"
+#import "SDLElectronicParkBrakeStatus.h"
 #import "SDLMyKey.h"
-#import "SDLRPCParameterNames.h"
-#import "SDLRPCFunctionNames.h"
 #import "SDLTireStatus.h"
+#import "SDLWiperStatus.h"
+#import "SDLECallInfo.h"
+#import "SDLVehicleDataEventStatus.h"
+#import "SDLDeviceStatus.h"
+#import "SDLPRNDL.h"
+#import "SDLClusterModeStatus.h"
+#import "SDLBeltStatus.h"
+#import "SDLAirbagStatus.h"
+#import "SDLGPSData.h"
+#import "SDLComponentVolumeStatus.h"
+#import "SDLFuelRange.h"
+#import "SDLRPCFunctionNames.h"
+#import "SDLRPCParameterNames.h"
```
These were changed to be out of order, please fix

joeljfischer commented on SmartDeviceLink/SDLOnVehicleData.h:
```diff
@@ -1,190 +1,216 @@
+#import "SDLComponentVolumeStatus.h"
 
-@class SDLAirbagStatus;
-@class SDLBeltStatus;
+@class SDLHeadLampStatus;
 @class SDLBodyInformation;
-@class SDLClusterModeStatus;
-@class SDLDeviceStatus;
+@class SDLMyKey;
+@class SDLTireStatus;
 @class SDLECallInfo;
+@class SDLDeviceStatus;
+@class SDLClusterModeStatus;
+@class SDLBeltStatus;
+@class SDLAirbagStatus;
+@class SDLGPSData;
 @class SDLEmergencyEvent;
 @class SDLFuelRange;
-@class SDLGPSData;
-@class SDLHeadLampStatus;
-@class SDLMyKey;
-@class SDLTireStatus;
```
These were changed to be out of order, please fix

joeljfischer commented on SmartDeviceLink/SDLOnVehicleData.h:
```diff
@@ -1,190 +1,216 @@
-
-NS_ASSUME_NONNULL_BEGIN
-
 @interface SDLOnVehicleData : SDLRPCNotification
```
When generating the file an initializer should have been created. Please add.

yoooriii replied:
Yes sure it was even though it looked funny. The initializer added. Also the unit test is updated to test this initializer.

joeljfischer commented on SmartDeviceLink/SDLOnVehicleData.m:
```diff
@@ -1,23 +1,57 @@
-#import "SDLDeviceStatus.h"
-#import "SDLECallInfo.h"
 #import "SDLEmergencyEvent.h"
-#import "SDLFuelRange.h"
-#import "SDLGPSData.h"
-#import "SDLHeadLampStatus.h"
+#import "SDLTurnSignal.h"
+#import "SDLElectronicParkBrakeStatus.h"
 #import "SDLMyKey.h"
-#import "SDLRPCParameterNames.h"
-#import "SDLRPCFunctionNames.h"
 #import "SDLTireStatus.h"
+#import "SDLWiperStatus.h"
+#import "SDLECallInfo.h"
+#import "SDLVehicleDataEventStatus.h"
+#import "SDLDeviceStatus.h"
+#import "SDLPRNDL.h"
+#import "SDLClusterModeStatus.h"
+#import "SDLBeltStatus.h"
+#import "SDLAirbagStatus.h"
+#import "SDLGPSData.h"
+#import "SDLComponentVolumeStatus.h"
+#import "SDLFuelRange.h"
+#import "SDLRPCFunctionNames.h"
+#import "SDLRPCParameterNames.h"
```
These were changed to be out of order, please fix

joeljfischer commented on SmartDeviceLink/SDLSubscribeVehicleData.h:
```diff
@@ -1,268 +1,386 @@
+ * @param handsOffSteering  handsOffSteering
+ * @return A SDLSubscribeVehicleData object
+ */
+- (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPosition airbagStatus:(BOOL)airbagStatus beltStatus:(BOOL)beltStatus bodyInformation:(BOOL)bodyInformation cloudAppVehicleID:(BOOL)cloudAppVehicleID clusterModeStatus:(BOOL)clusterModeStatus deviceStatus:(BOOL)deviceStatus driverBraking:(BOOL)driverBraking eCallInfo:(BOOL)eCallInfo electronicParkBrakeStatus:(BOOL)electronicP ...
```
When generating the file, the initializer that's created should use `nullable` `NSNumber`s. Also, the new parameter should be in alphabetical order.

joeljfischer commented on SmartDeviceLink/SDLSubscribeVehicleDataResponse.h:
```diff
@@ -1,19 +1,44 @@
-
-NS_ASSUME_NONNULL_BEGIN
-
 @interface SDLSubscribeVehicleDataResponse : SDLRPCResponse
```
When generating the file an initializer should have been created, please add.

joeljfischer commented on SmartDeviceLink/SDLSubscribeVehicleDataResponse.m:
```diff
@@ -1,13 +1,40 @@
 #import "SDLSubscribeVehicleDataResponse.h"
-
 #import "NSMutableDictionary+Store.h"
-#import "SDLRPCParameterNames.h"
-#import "SDLRPCFunctionNames.h"
 #import "SDLVehicleDataResult.h"
+#import "SDLRPCFunctionNames.h"
+#import "SDLRPCParameterNames.h"
```
Changed to be out of order, please fix

joeljfischer commented on SmartDeviceLink/SDLSubscribeVehicleData.h:
```diff
@@ -1,268 +1,386 @@
+ * @param wiperStatus Subscribe to wiperStatus
+ * @return A SDLSubscribeVehicleData object
+ */
+- (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPosition airbagStatus:(BOOL)airbagStatus beltStatus:(BOOL)beltStatus bodyInformation:(BOOL)bodyInformation clusterModeStatus:(BOOL)clusterModeStatus deviceStatus:(BOOL)deviceStatus driverBraking:(BOOL)driverBraking eCallInfo:(BOOL)eCallInfo emergencyEvent:(BOOL)emergencyEvent engineTorque:(BOOL)engineTorque externalTemperatu ...
```
The deprecation notice is missing the new parameter

joeljfischer commented on SmartDeviceLink/SDLSubscribeVehicleData.h:
```diff
@@ -1,268 +1,386 @@
+ * @param wiperStatus Subscribe to wiperStatus
+ * @return A SDLSubscribeVehicleData object
+ */
+- (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPosition airbagStatus:(BOOL)airbagStatus beltStatus:(BOOL)beltStatus bodyInformation:(BOOL)bodyInformation clusterModeStatus:(BOOL)clusterModeStatus deviceStatus:(BOOL)deviceStatus driverBraking:(BOOL)driverBraking eCallInfo:(BOOL)eCallInfo electronicParkBrakeStatus:(BOOL)electronicParkBrakeStatus emergencyEvent:(BOOL)emerge ...
```
The deprecation notice is missing the new parameter

joeljfischer commented on SmartDeviceLink/SDLUnsubscribeVehicleData.h:
```diff
@@ -1,270 +1,385 @@
+ *  @param handsOffSteering - handsOffSteering
+ *  @return A SDLUnsubscribeVehicleData object
+ */
+- (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPosition airbagStatus:(BOOL)airbagStatus beltStatus:(BOOL)beltStatus bodyInformation:(BOOL)bodyInformation  cloudAppVehicleID:(BOOL)cloudAppVehicleID clusterModeStatus:(BOOL)clusterModeStatus deviceStatus:(BOOL)deviceStatus driverBraking:(BOOL)driverBraking eCallInfo:(BOOL)eCallInfo electronicParkBrakeStatus:(BOOL)electronic ...
```
See above about `NSNumber`s

yoooriii replied:
For now 2 new initializers added: a manually created one and a script generated. Please advice which one to use.

joeljfischer commented on SmartDeviceLink/SDLUnsubscribeVehicleDataResponse.h:
```diff
@@ -1,17 +1,43 @@
- since SmartDeviceLink 2.0
+ * @since SDL 2.0.0
  */
 @interface SDLUnsubscribeVehicleDataResponse : SDLRPCResponse
```
See above about adding initializer

yoooriii replied:
Script generated initializer added to both *.h and *.m files

joeljfischer commented on SmartDeviceLink/SDLUnsubscribeVehicleDataResponse.m:
```diff
@@ -1,13 +1,40 @@
 #import "SDLUnsubscribeVehicleDataResponse.h"
-
 #import "NSMutableDictionary+Store.h"
-#import "SDLRPCParameterNames.h"
-#import "SDLRPCFunctionNames.h"
 #import "SDLVehicleDataResult.h"
+#import "SDLRPCFunctionNames.h"
+#import "SDLRPCParameterNames.h"
```
Out of order

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
-        expect(testRequest.tirePressure).to(beNil());
-        expect(testRequest.turnSignal).to(beNil());
-        expect(testRequest.wiperStatus).to(beNil());
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters to nil", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
-        SDLGetVehicleData *testRequest = [[SDLGetVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:NO beltStatus:YES bodyInformation:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES emergencyEvent:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState:YES gps:YES headLampStatus:YES instantFuelConsumption:YES myKey:YES odometer:YES p ...
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    SDLGetVehicleData *testRequest = [[SDLGetVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:NO beltStatus:YES bodyInformation:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES emergencyEvent:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState:YES gps:YES headLampStatus:YES instantFuelConsumption:YES myKey:YES odometer:YES prndl ...
```
```suggestion
        SDLGetVehicleData *testRequest = [[SDLGetVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:NO beltStatus:YES bodyInformation:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES emergencyEvent:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState:YES gps:YES headLampStatus:YES instantFuelConsumption:YES myKey:YES odometer:YES prndl:YES rpm:YES speed:YES steeringWheelAngle:YES tirePressure:YES vin:YES wiperStatus:YES];
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
-        expect(testRequest.turnSignal).to(equal(@NO));
-        expect(testRequest.wiperStatus).to(equal(@YES));
-        #pragma clang diagnostic pop
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
-        expect(testRequest.turnSignal).to(equal(@YES));
-        expect(testRequest.wiperStatus).to(equal(@YES));
-        #pragma clang diagnostic pop
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
-        expect(testRequest.tirePressure).to(equal(@YES));
-        expect(testRequest.turnSignal).to(equal(@YES));
-        expect(testRequest.wiperStatus).to(equal(@YES));
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
+
+    context(@"INIT-4", ^{
+        SDLGetVehicleData *testRequest =
+    [[SDLGetVehicleData alloc] initWithAccelerationPedalPosition:YES
+                                                    airbagStatus:YES
+                                                      beltStatus:YES
+                                                 bodyInformation:YES
+                                               cloudAppVehicleID:YES
+                                               clusterModeStatus:YES
+                                                    deviceStatus:YES
+                                                   driverBraking:YES
+                                                       eCallInfo:YES
+                                       electronicParkBrakeStatus:YES
+                                                  emergencyEvent:YES
+                                                   engineOilLife:YES
+                                                    engineTorque:YES
+                                             externalTemperature:YES
+                                                       fuelLevel:YES
+                                                  fuelLevelState:YES
+                                                       fuelRange:YES
+                                                             gps:YES
+                                                  headLampStatus:YES
+                                          instantFuelConsumption:YES
+                                                           myKey:YES
+                                                        odometer:YES
+                                                           prndl:YES
+                                                             rpm:YES
+                                                           speed:YES
+                                              steeringWheelAngle:YES
+                                                    tirePressure:YES
+                                                      turnSignal:YES
+                                                             vin:YES
+                                                     wiperStatus:YES
+                                                handsOffSteering:YES];
```
Just do this on one line

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -146,149 +150,232 @@
+                                                             vin:YES
+                                                     wiperStatus:YES
+                                                handsOffSteering:YES];
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:
```diff
@@ -297,11 +384,11 @@
-        expect([testRequest getOEMCustomVehicleData:@"OEMCustomVehicleData"]).to(beFalse());
-        expect([testRequest getOEMCustomVehicleData:@"OEMCustomVehicleData1"]).to(beTrue());
-
+        it(@"all set", ^{
```
```suggestion
        it(@"should set the custom data properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
-        expect(testRequest.tirePressure).to(beNil());
-        expect(testRequest.turnSignal).to(beNil());
-        expect(testRequest.wiperStatus).to(beNil());
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters to nil", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
 
-    context(@"initWithAccelerationPedalPosition:airbagStatus:beltStatus:bodyInformation:cloudAppVehicleID:clusterModeStatus:deviceStatus:driverBraking:eCallInfo:emergencyEvent:engineOilLife:engineTorque:externalTemperature:fuelLevel:fuelLevelState:gps:headLampStatus:instantFuelConsumption:myKey:odometer:prndl:rpm:speed:steeringWheelAngle:tirePressure:wiperStatus:", ^{
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:YES beltStatus:YES bodyInformation:YES cloudAppVehicleID:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES electronicParkBrakeStatus:YES emergencyEvent:YES engineOilLife:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState ...
+context(@"initWithAccelerationPedalPosition:airbagStatus:beltStatus:bodyInformation:cloudAppVehicleID:clusterModeStatus:deviceStatus:driverBraking:eCallInfo:emergencyEvent:engineOilLife:engineTorque:externalTemperature:fuelLevel:fuelLevelState:gps:headLampStatus:instantFuelConsumption:myKey:odometer:prndl:rpm:speed:steeringWheelAngle:tirePressure:wiperStatus:", ^{
```
Indentation is incorrect

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
-        expect(testRequest.tirePressure).to(equal(@YES));
-        expect(testRequest.turnSignal).to(equal(@YES));
-        expect(testRequest.wiperStatus).to(equal(@YES));
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

yoooriii replied:
"should set all the parameters to YES" - maybe it is better to be more specific

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
-        expect(testRequest.turnSignal).to(equal(@YES));
-        expect(testRequest.wiperStatus).to(equal(@YES));
-        #pragma clang diagnostic pop
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:
```diff
@@ -146,160 +150,172 @@
-        expect([testRequest getOEMCustomVehicleData:@"customVehicleData"]).to(beFalse());
-        expect([testRequest getOEMCustomVehicleData:@"customVehicleData1"]).to(beTrue());
-
+        it(@"all set", ^{
```
```suggestion
        it(@"should set the custom data properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
-        expect(testRequest.tirePressure).to(beNil());
-        expect(testRequest.turnSignal).to(beNil());
-        expect(testRequest.wiperStatus).to(beNil());
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters to nil", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
     });
     context(@"initWithAccelerationPedalPosition:airbagStatus:beltStatus:bodyInformation:cloudAppVehicleID:clusterModeStatus:deviceStatus:driverBraking:eCallInfo:electronicParkBrakeStatus:emergencyEvent:engineOilLife:engineTorque:externalTemperature:fuelLevel:fuelLevelState:gps:headLampStatus:instantFuelConsumption:myKey:odometer:prndl:rpm:speed:steeringWheelAngle:tirePressure:turnSignal:wiperStat ...
-        SDLUnsubscribeVehicleData *testRequest = [[SDLUnsubscribeVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:YES beltStatus:YES bodyInformation:YES cloudAppVehicleID:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES electronicParkBrakeStatus:YES emergencyEvent:YES engineOilLife:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelS ...
+    SDLUnsubscribeVehicleData *testRequest = [[SDLUnsubscribeVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:YES beltStatus:YES bodyInformation:YES cloudAppVehicleID:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES electronicParkBrakeStatus:YES emergencyEvent:YES engineOilLife:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState ...
```
```suggestion
        SDLUnsubscribeVehicleData *testRequest = [[SDLUnsubscribeVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:YES beltStatus:YES bodyInformation:YES cloudAppVehicleID:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES electronicParkBrakeStatus:YES emergencyEvent:YES engineOilLife:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState:YES fuelRange:YES gps:YES headLampStatus:YES instantFuelConsumption:YES myKey:YES odometer:YES prndl:YES rpm:YES speed:YES steeringWheelAngle:YES tirePressure:YES turnSignal:YES wiperStatus:YES handsOffSteering:YES];
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
-        expect(testRequest.tirePressure).to(equal(@YES));
-        expect(testRequest.turnSignal).to(equal(@YES));
-        expect(testRequest.wiperStatus).to(equal(@YES));
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

yoooriii replied:
I propose to be more specific: @"should set all the parameters to YES"

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
-        expect(testRequest.turnSignal).to(equal(@NO));
-        expect(testRequest.wiperStatus).to(equal(@YES));
-        #pragma clang diagnostic pop
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the parameters properly", ^{
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:
```diff
@@ -146,158 +150,171 @@
-        expect([testRequest getOEMCustomVehicleData:@"customVehicleData"]).to(beFalse());
-        expect([testRequest getOEMCustomVehicleData:@"customVehicleData1"]).to(beTrue());
-
+        it(@"all set", ^{
```
```suggestion
        it(@"should set all the custom data properly", ^{
```

yoooriii commented on SmartDeviceLink/SDLGetVehicleData.m:
```diff
@@ -69,6 +98,47 @@ - (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPositio
     return self;
 }
 
+- (instancetype)initWithAccelerationPedalPosition:(BOOL)accelerationPedalPosition airbagStatus:(BOOL)airbagStatus beltStatus:(BOOL)beltStatus bodyInformation:(BOOL)bodyInformation cloudAppVehicleID:(BOOL)cloudAppVehicleID clusterModeStatus:(BOOL)clusterModeStatus deviceStatus:(BOOL)deviceStatus driverBraking:(BOOL)driverBraking eCallInfo:(BOOL)eCallInfo electronicParkBrakeStatus:(BOOL)electronicP ...
```
I am afraid I did not get the idea, all the arguments in the initializers are BOOLs and as such they cannot be nullable, just in case I declared initializers nullable like this: -(nullable instancetype)initWithBool:(BOOL)boolValue;

yoooriii replied:
I have added the initializer generated with the RPC generator (the version from git@github.com:LuxoftSDL/rpc_spec.git; branch: /feature/0257_vehicle_data_handsOffSteering);

Should I reorder all arguments alphabetically or RPC generated methods take precedence ?

joeljfischer replied:
Thanks. How does the RPC generator order them? Based on the order in the RPC spec? If so, I think it's best to use the generated version.

yoooriii replied:
It looks like the script keeps the RPC spec (MOBILE_API.xml) order. Just in case I commited both versions (hand made and script generated) could you please review and advice which one we should use? Thank you.

joeljfischer replied:
As much as I don't like it what the generated code is, part of the RPC generator proposal was to use the generated types, so that is the one that should be used.

joeljfischer commented:
@yooorii I see that you requested a review, but you have not updated everything as asked and there are merge conflicts. Please fix before a review occurs.

yoooriii commented:
Hello, @joeljfischer. Could you please review my last changes? I have updated codebase as was requested and also updated the example app to handle 'hands off steering' data.

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:
```diff
@@ -219,9 +221,67 @@
+    vin = @"6574839201a";
+    cloudAppVehicleID = @"cloudAppVehicleID";
+
+    fuelRange = [[SDLFuelRange alloc] init];
+        SDLGetVehicleDataResponse *testResponse = [[SDLGetVehicleDataResponse alloc] initWithGps:gps speed:100.0 rpm:@3 fuelLevel:99.9999 fuelLevel_State:SDLComponentVolumeStatusFault instantFuelConsumption:40.7 fuelRange:@[fuelRange] externalTemperature:0.5 turnSignal:SDLTurnSignalOff vin:vin prndl:SDLPRNDLPark tirePressure:tires odometer:@7000 beltStatus:belt bodyInformation:body deviceStatus:d ...
+
+
+        it(@"Should get correctly when initialized", ^{
+            expect(testResponse.accPedalPosition).to(equal(@0.5));
+            expect(testResponse.airbagStatus).to(equal(airbag));
+            expect(testResponse.beltStatus).to(equal(belt));
+            expect(testResponse.bodyInformation).to(equal(body));
+            expect(testResponse.cloudAppVehicleID).to(equal(cloudAppVehicleID));
+            expect(testResponse.clusterModeStatus).to(equal(clusterMode));
+            expect(testResponse.deviceStatus).to(equal(device));
+            expect(testResponse.driverBraking).to(equal(SDLVehicleDataEventStatusNoEvent));
+            expect(testResponse.eCallInfo).to(equal(eCall));
+            expect(testResponse.electronicParkBrakeStatus).to(equal(SDLElectronicParkBrakeStatusDriveActive));
+            expect(testResponse.emergencyEvent).to(equal(event));
+            expect(testResponse.engineOilLife).to(equal(@(23.22f)));
+            expect(testResponse.engineTorque).to(equal(@(630.4f)));
+            expect(testResponse.externalTemperature).to(equal(@0.5));
+            expect(testResponse.fuelLevel).to(equal(@(99.9999f)));
+            expect(testResponse.fuelLevel_State).to(equal(SDLComponentVolumeStatusFault));
+            expect(testResponse.fuelRange).to(equal(@[fuelRange]));
+            expect(testResponse.gps).to(equal(gps));
+            expect(testResponse.headLampStatus).to(equal(headLamp));
+            expect(testResponse.instantFuelConsumption).to(equal(@(40.7f)));
+            expect(testResponse.myKey).to(equal(myKey));
+            expect(testResponse.odometer).to(equal(@(7000.0f)));
+            expect(testResponse.prndl).to(equal(SDLPRNDLPark));
+            expect(testResponse.rpm).to(equal(@3));
+            expect(testResponse.speed).to(equal(@100));
+            expect(testResponse.steeringWheelAngle).to(equal(@-1500));
+            expect(testResponse.tirePressure).to(equal(tires));
+            expect(testResponse.turnSignal).to(equal(SDLTurnSignalOff));
+            expect(testResponse.vin).to(equal(vin));
+            expect(testResponse.wiperStatus).to(equal(SDLWiperStatusAutomaticHigh));
+            expect(testResponse.handsOffSteering).to(equal(@YES));
+        });
```
These don't seem to be aligned correctly

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:
```diff
@@ -198,4 +201,45 @@
+describe(@"Convenience init test", ^ {
+    context(@"initWithGps:speed:rpm:fuelLevel:fuelLevel_State:instantFuelConsumption:fuelRange:externalTemperature:turnSignal:prndl:tirePressure:odometer:beltStatus:bodyInformation:deviceStatus:driverBraking:wiperStatus:headLampStatus:engineTorque:accPedalPosition:steeringWheelAngle:engineOilLife:electronicParkBrakeStatus:cloudAppVehicleID:eCallInfo:airbagStatus:emergencyEvent:clusterModes:myKey: ...
+        SDLSubscribeVehicleDataResponse *testResponse = [[SDLSubscribeVehicleDataResponse alloc] initWithGps:vehicleDataResult speed:vehicleDataResult rpm:vehicleDataResult fuelLevel:vehicleDataResult fuelLevel_State:vehicleDataResult instantFuelConsumption:vehicleDataResult fuelRange:vehicleDataResult externalTemperature:vehicleDataResult turnSignal:vehicleDataResult prndl:vehicleDataResult tire ...
+        [testResponse setOEMCustomVehicleData:@"customOEMVehicleData" withVehicleDataState:customOEMvehicleDataResult];
```
```suggestion
```

joeljfischer commented on SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:
```diff
@@ -198,4 +201,45 @@
+        [testResponse setOEMCustomVehicleData:@"customOEMVehicleData" withVehicleDataState:customOEMvehicleDataResult];
+
+        it(@"Expect all the params to be vehicleDataResult", ^{
+            expect([testResponse getOEMCustomVehicleData:@"customOEMVehicleData"]).to(equal(customOEMvehicleDataResult));
```
```suggestion
```

yoooriii commented on SmartDeviceLink-iOS.xcodeproj/xcshareddata/xcschemes/SmartDeviceLink-Example-ObjC.xcscheme:
```diff
@@ -28,6 +28,16 @@
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       shouldUseLaunchSchemeArgsEnv = "YES">
       <Testables>
+         <TestableReference
+            skipped = "NO">
+            <BuildableReference
+               BuildableIdentifier = "primary"
+               BlueprintIdentifier = "5D61FA251A84237100846EE7"
+               BuildableName = "SmartDeviceLinkTests.xctest"
+               BlueprintName = "SmartDeviceLinkTests"
+               ReferencedContainer = "container:SmartDeviceLink-iOS.xcodeproj">
+            </BuildableReference>
+         </TestableReference>
```
reverted

yoooriii commented:
Hello @joeljfischer , could you please review the latest change? Thank you.

yoooriii commented:
Hello @joeljfischer, could you please review? I have done the last 2 minor changes. Thank you.

joeljfischer commented:
Hello @yoooriii, this PR has been approved, can you please ensure that the type of changes in other PRs are consistent with this one and advise when you believe each is ready for review? Thank you.

yoooriii commented:
Thank you @joeljfischer. Then I will finalize the rest PRs ASAP, hopefully by Monday.
